lib/compat/wordpress-6.7/blocks.php

```diff
@@ -43,3 +43,58 @@ function gutenberg_filter_block_type_metadata_settings_allow_variations_php_file
 	return $settings;
 }
 add_filter( 'block_type_metadata_settings', 'gutenberg_filter_block_type_metadata_settings_allow_variations_php_file', 10, 2 );
+
+/**
+ * Filter the query vars (tax_query) for the Query Loop block to support post formats.
+ *
+ * @param array    $query The query vars.
+ * @param WP_Block $block Block instance.
+ * @return array   The filtered query vars.
+ */
+function gutenberg_filter_query_loop_block_query_vars_post_format( $query, $block ) {
+	if ( isset( $block->context['query']['format'] ) &&
+		! empty( $block->context['query']['format'] ) ) {
+
+		$formats   = $block->context['query']['format'];
+		$tax_query = array( 'relation' => 'OR' );
+
+		// The default post format, 'standard', is not saved in the database.
+		// If 'standard' is part of the request, the query needs to exclude the formats.
+		if ( in_array( 'standard', $formats, true ) ) {
+			$tax_query[] = array(
+				'taxonomy' => 'post_format',
+				'field'    => 'slug',
+				'terms'    => array(),
+				'operator' => 'NOT EXISTS',
+			);
+			// Remove the standard format:
+			$formats = array_diff( $formats, array( 'standard' ) );
+		}
```

Comment:
I find the comments here a bit confusing. It seems like we're saying that a query with standard _must_ list it as excluded, but then assuming also that if it's in here, it _is_ excluded.

Does this handle cases where someone requests _including_ `standard`?

Reply:
Standard includes all items that do not have a post format assigned.
It excludes all items that do have a post format assigned.

Standard can not be searched for in the database, that is why it is removed from the query.

When the user selects both standard and a format, they are combined so that you can query for both
"video" and "no format assigned".

Reply:
How about

```
	// The default post format, 'standard', is not stored in the database.
	// If 'standard' is part of the request, the query needs to exclude all post items that
	// has a format assigned.
	if ( in_array( 'standard', $formats, true ) ) {
		$tax_query[] = array(
			'taxonomy' => 'post_format',
			'field'    => 'slug',
			'terms'    => array(),
			'operator' => 'NOT EXISTS',
		);
		// Remove the standard format, since it cannot be queried.
		$formats = array_diff( $formats, array( 'standard' ) );
	}
```

Reply:
the comment here is much clearer. thank you.

small nitpick: there's a correspondence typo in "all post items that <del>has</del><ins>have</ins> a format assigned."